src/CMakeLists.txt

```diff
@@ -246,17 +230,151 @@ ADD_DEFINITIONS(-DEIGEN_MPL2_ONLY)
 # - external if CERES_DIR_HINTS and FIND_PACKAGE return a valid Ceres setup
 # ==============================================================================
 FIND_PACKAGE(Ceres QUIET HINTS ${CERES_DIR_HINTS})
-IF(Ceres_FOUND)
-   MESSAGE("CERES FOUND")
-ELSE()
-  MESSAGE("CERES NOT FOUND")
+IF(NOT Ceres_FOUND)
   SET(OpenMVG_USE_INTERNAL_CERES ON)
   SET(CERES_INCLUDE_DIRS
     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/ceres-solver/include
     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/ceres-solver/internal/ceres/miniglog
     ${PROJECT_BINARY_DIR}/third_party/ceres-solver/config)
+    FILE(READ "${CMAKE_CURRENT_SOURCE_DIR}/third_party/ceres-solver/VERSION" CERES_CONFIG)
+    STRING(REGEX REPLACE "version ([0-9.]+).*" "\\1" CERES_VERSION ${CERES_CONFIG})
   SET(CERES_LIBRARIES ceres cxsparse)
-ENDIF(Ceres_FOUND)
+ENDIF()
+
+# ==============================================================================
+# Flann
+# ==============================================================================
+# - internal by default (flann),
+# - external if FLANN_DIR_HINTS and a valid Flann setup is found
+# ==============================================================================
+FIND_PATH(FLANN_INCLUDE_DIRS flann.hpp ${FLANN_DIR_HINTS})
```

Comment:
I'am not an expert but isn't it better to pass through a `find_package()` instead of looking for the `.h` of each library inside the `LIBNAME_DIR_HINTS`? It seems to me that this can work only if the library that you are looking for are installed in the usual system directories (eg /usr/lib): when u later call `FIND_LIBRARY(LIBNAME_LIBRARY NAMES libname)` you don't provide any other directory in which to look for. If the library is installed in a non system path there's no way it could look into it.
For example, if have installed flann in some directory in my home `/home/simone/my/personal/flann/install/path/` then according to this version I should pass `FLANN_DIR_HINTS=/home/simone/my/personal/flann/install/path/include/flann` (where the `flann.hpp` is installed) but then the `FIND_LIBRARY(FLANN_LIBRARY NAMES flann_cpp)` will have no clue that the .so are placed somewhere inside `/home/simone/my/personal/flann/install/path/lib`.